iOS/FVM/FVM/FVMCarModelViewController.swift

```diff
@@ -81,17 +62,19 @@ open class FVMCarModelViewController : SCNView {
         let hitResults = self.hitTest(p, options: [:])
         if hitResults.count > 0 {
             let result = hitResults.first!
-            if highlightedParts.contains(result.node) {
+            let highlightedNodes = highlightedParts.map { $0.0 }
+            if highlightedNodes.contains(result.node) {
                 SCNTransaction.begin()
                 SCNTransaction.animationDuration = 0.5
-                highlightedParts.remove(at: highlightedParts.firstIndex(of: result.node)!)
-                result.node.scale = SCNVector3(x: 1, y: 1, z: 1)
+                let hitPartIndex = highlightedNodes.firstIndex(of: result.node)
+                result.node.geometry?.firstMaterial?.diffuse.contents = highlightedParts[hitPartIndex!].material
+                highlightedParts.remove(at: highlightedNodes.firstIndex(of: result.node)!)
                 SCNTransaction.commit()
```

Comment:
Code duplication:
`SCNTransaction.begin()
                 SCNTransaction.animationDuration = 0.5		                 SCNTransaction.animationDuration = 0.5`

...
`SCNTransaction.commit()`

Reply:
It's seems that Controller class does everything. We have to think how to move some responsibilities to another class.

Reply:
I'd create US for this. For now, leaving as it is.